models/CMakeLists.txt

```diff
@@ -18,127 +18,14 @@
-    erfc_neuron.h erfc_neuron.cpp
-    gamma_sup_generator.h gamma_sup_generator.cpp
-    gap_junction.h
-    gauss_rate.h gauss_rate.cpp
-    gif_psc_exp.h gif_psc_exp.cpp
-    gif_psc_exp_multisynapse.h gif_psc_exp_multisynapse.cpp
-    gif_cond_exp.h gif_cond_exp.cpp
-    gif_cond_exp_multisynapse.h gif_cond_exp_multisynapse.cpp
-    gif_pop_psc_exp.h gif_pop_psc_exp.cpp
-    ginzburg_neuron.h ginzburg_neuron.cpp
-    glif_cond.h glif_cond.cpp
-    glif_psc.h glif_psc.cpp
-    hh_cond_exp_traub.h hh_cond_exp_traub.cpp
-    hh_cond_beta_gap_traub.h hh_cond_beta_gap_traub.cpp
-    hh_psc_alpha.h hh_psc_alpha.cpp
-    hh_psc_alpha_clopath.h hh_psc_alpha_clopath.cpp
-    hh_psc_alpha_gap.h hh_psc_alpha_gap.cpp
-    ht_synapse.h
-    ht_neuron.h ht_neuron.cpp
-    iaf_chs_2007.cpp iaf_chs_2007.h
-    iaf_chxk_2008.cpp iaf_chxk_2008.h
-    iaf_cond_alpha.h iaf_cond_alpha.cpp
-    iaf_cond_alpha_mc.h iaf_cond_alpha_mc.cpp
-    iaf_cond_beta.h iaf_cond_beta.cpp
-    iaf_cond_exp.h iaf_cond_exp.cpp
-    iaf_cond_exp_sfa_rr.h iaf_cond_exp_sfa_rr.cpp
-    iaf_psc_alpha.h iaf_psc_alpha.cpp
-    iaf_psc_alpha_multisynapse.h iaf_psc_alpha_multisynapse.cpp
-    iaf_psc_alpha_ps.cpp iaf_psc_alpha_ps.h
-    iaf_psc_delta.h iaf_psc_delta.cpp
-    iaf_psc_delta_ps.cpp iaf_psc_delta_ps.h
-    iaf_psc_exp.h iaf_psc_exp.cpp
-    iaf_psc_exp_htum.h iaf_psc_exp_htum.cpp
-    iaf_psc_exp_multisynapse.h iaf_psc_exp_multisynapse.cpp
-    iaf_psc_exp_ps.cpp iaf_psc_exp_ps.h
-    iaf_psc_exp_ps_lossless.cpp iaf_psc_exp_ps_lossless.h
-    izhikevich.h izhikevich.cpp
-    jonke_synapse.h
-    lin_rate.h lin_rate.cpp
-    mat2_psc_exp.h mat2_psc_exp.cpp
-    mcculloch_pitts_neuron.h mcculloch_pitts_neuron.cpp
-    mip_generator.h mip_generator.cpp
-    modelsmodule.h modelsmodule.cpp
-    multimeter.h multimeter.cpp
-    music_cont_in_proxy.h music_cont_in_proxy.cpp
-    music_cont_out_proxy.h music_cont_out_proxy.cpp
-    music_event_in_proxy.h music_event_in_proxy.cpp
-    music_event_out_proxy.h music_event_out_proxy.cpp
-    music_rate_in_proxy.h music_rate_in_proxy.cpp
-    music_rate_out_proxy.h music_rate_out_proxy.cpp
-    music_message_in_proxy.h music_message_in_proxy.cpp
-    noise_generator.h noise_generator.cpp
-    parrot_neuron.h parrot_neuron.cpp
-    parrot_neuron_ps.cpp parrot_neuron_ps.h
-    inhomogeneous_poisson_generator.h inhomogeneous_poisson_generator.cpp
-    poisson_generator.h poisson_generator.cpp
-    poisson_generator_ps.cpp poisson_generator_ps.h
-    pp_psc_delta.h pp_psc_delta.cpp
-    pp_cond_exp_mc_urbanczik.h pp_cond_exp_mc_urbanczik.cpp
-    ppd_sup_generator.h ppd_sup_generator.cpp
-    pulsepacket_generator.h pulsepacket_generator.cpp
-    quantal_stp_synapse.h quantal_stp_synapse_impl.h
-    rate_connection_delayed.h
-    rate_connection_instantaneous.h
-    rate_neuron_opn.h rate_neuron_opn_impl.h
+    cm_tree.h cm_tree.cpp
     rate_neuron_ipn.h rate_neuron_ipn_impl.h
+    rate_neuron_opn.h rate_neuron_opn_impl.h
```

Comment:
Why are these still hard-coded?

Reply:
These are the auxiliary helper/base classes we talked about the other day. They need compilation/inclusion (see [lines 48-50](https://github.com/nest/nest-simulator/pull/2835/files/ae884a60c9546a64b8ac8f79d1e5473bfaafe4b0#diff-d0e03f64b9b96929ef3763efbeca17202c027ecf72059b6a12a676aefee1a372R48-R50) but don't define any models themselves. I thought the easiest way was to include them here irregardless of whether they are needed or not and assume that a modern compiler is smart enough to leave their symbols/definitions out of the resulting executable if they are not needed.

Reply:
I understand, but why can't we anyway just put them in the "rate" (and "full") modelsets?

Reply:
Because then 
1. users who want to build NEST with certain custom modelset would have to know about these (purely technical dependencies), i.e. `-Dwith-models=gauss_rate,step_rate_generator,multimeter` would also have to include `rate_ipn`, `rate_opn` and `rate_transformer_node` (and in the future possible others).
2. the generation pipeline would potentially have to treat the auxiliary files differently, as they don't actually define any models